pkg/statsd/statsd.go

```diff
@@ -118,14 +119,32 @@ func (s *Server) RunWithCustomSocket(ctx context.Context, sf SocketFactory) erro
 	})
 
 	// 4. Start the heartbeat
-
 	if s.HeartbeatInterval != 0 {
 		hb := stats.NewHeartBeater(statser, "heartbeat", s.HeartbeatTags, s.HeartbeatInterval)
 		stage = stgr.NextStage()
 		stage.StartWithContext(hb.Run)
 	}
 
-	// 5. Start the Receiver
+	// 5. Start the Parser
+	// Open receiver <-> parser chan
+	datagrams := make(chan *Datagram, s.MaxReaders*s.ReceiveBatchSize)
+	dgStats := stats.NewChannelStatsWatcher(
+		statser,
+		"statsd.channel",
```

Comment:
This is the channel name, not the metric name.  It ends up as the channel tag against the <namespace.>channel.* metric.

Reply:
Also METRICS.md, there's a section just for CSWs

Reply:
right yeah sorry, I was looking at Alexander's implementation